src/transaction.js

```diff
@@ -4,9 +4,15 @@ var bufferutils = require('./bufferutils')
 var opcodes = require('./opcodes')
 var typeforce = require('typeforce')
 var types = require('./types')
+var BufferWriter = require('./bufferwriter')
+
+var ADVANCED_TRANSACTION_MARKER = 0
+var ADVANCED_TRANSACTION_FLAG = 1
 
 function Transaction () {
   this.version = 1
+  this.marker = null
+  this.flag = null
```

Comment:
The zero marker makes it obvious to old clients that something is wrong (zero-inputs), and I don't think it will change. Flags may change - it's interpreted as a bit-field where the first indicates segwit support, but setting anything else will intentionally result in an error. https://github.com/bitcoin/bitcoin/blob/2f71490d21796594ca6f55e375558944de9db5a0/src/primitives/transaction.h#L320